### PR TITLE
IGNITE-18574

### PR DESCRIPTION
**Pull request by maksaska**
* Kafka consumer/producer timeouts documentation

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: docs/_docs/extensions-and-integrations/change-data-capture-extensions.adoc [hunk 1]
```diff
@@ -118,6 +118,11 @@ image:../../assets/images/integrations/CDC-ignite2kafka.svg[]
 | `kafkaRequestTimeout` | Kafka request timeout in milliseconds.  | `3000`
 |===
 
+`kafkaRequestTimeout` property sets how much `IgniteToKafkaCdcStreamer` will wait for `KafkaProducer` to finish request. Also, proper `KafkaProducer` configuration should be provided, including link:https://kafka.apache.org/documentation/#producerconfigs_request.timeout.ms[request.timeout.ms]. For more details you should refer to a link:https://kafka.apache.org/documentation/#configuration[configuration]
```
Comment (timoninmaxim) [inline docs/_docs/extensions-and-integrations/change-data-capture-extensions.adoc:121]: Docs should describe how to configure `KafkaProducer`.

File: docs/_docs/extensions-and-integrations/change-data-capture-extensions.adoc [hunk 2]
```diff
@@ -118,6 +118,11 @@ image:../../assets/images/integrations/CDC-ignite2kafka.svg[]
...
+section of the official Kafka documentation.
+
+NOTE: `kafkaRequestTimeout` should not be too low, because `IgniteToKafkaCdcStreamer` will wait for `KafkaProducer` to send messages to a broker, and if wait time exceeds `kafkaRequestTimeout`, then `IgniteToKafkaCdcStreamer` will fail with a timeout error.
```
Comment (timoninmaxim) [inline docs/_docs/extensions-and-integrations/change-data-capture-extensions.adoc:124]: > kafkaRequestTimeout` property sets how much `IgniteToKafkaCdcStreamer` will wait for `KafkaProducer` to finish request

and

> `IgniteToKafkaCdcStreamer` will wait for `KafkaProducer` to send messages to a broker

Duplication of the same idea.

File: docs/_docs/extensions-and-integrations/change-data-capture-extensions.adoc [hunk 3]
```diff
@@ -180,11 +185,20 @@ Kafka to Ignite configuration file should contain the following beans that will
...
 | `metadataConsumerGroup` | Group for `KafkaConsumer`, which polls from metadata topic | ignite-metadata-update-<kafkaPartsFrom>-<kafkaPartsTo>
 | `kafkaRequestTimeout` | Kafka request timeout in milliseconds.  | `3000`
+| `kafkaConsumerPollTimeout` | Kafka poll timeout in milliseconds. | `3000`
 | `maxBatchSize` | Maximum number of events to be sent to destination cluster in a single batch. | 1024
 | `threadCount` | Count of threads to proceed consumers. Each thread poll records from dedicated partitions in round-robin manner. | 16
 |`metricRegistryName`| Name for metric registry. `org.apache.metricRegistryName.cdc.applier` | cdc-kafka-to-ignite
 |===
 
+`kafkaRequestTimeout` property is used as timeout for `KafkaConsumer` methods (except for `KafkaConsumer#poll`). Also, proper `KafkaConsumer` configuration should be provided, including link:https://kafka.apache.org/documentation/#consumerconfigs_request.timeout.ms[request.timeout.ms]. For more details you should refer to a link:https://kafka.apache.org/documentation/#configuration[configuration] section of the official Kafka documentation.
```
Comment (timoninmaxim) [inline docs/_docs/extensions-and-integrations/change-data-capture-extensions.adoc:194]: same for `KafkaConsumer`

**Review by timoninmaxim — CHANGES_REQUESTED** [review]

File: docs/_docs/extensions-and-integrations/change-data-capture-extensions.adoc (hunk 1 above)
Comment (maksaska) [reply docs/_docs/extensions-and-integrations/change-data-capture-extensions.adoc]: Good point!

File: docs/_docs/extensions-and-integrations/change-data-capture-extensions.adoc (hunk 2 above)
Comment (maksaska) [reply docs/_docs/extensions-and-integrations/change-data-capture-extensions.adoc]: Agree!

File: docs/_docs/extensions-and-integrations/change-data-capture-extensions.adoc (hunk 3 above)
Comment (maksaska) [reply docs/_docs/extensions-and-integrations/change-data-capture-extensions.adoc]: Yes, changed it

**Review by timoninmaxim — APPROVED** [review]

**Pull request merged**